tests/unit/commands/local/lib/test_cfn_api_provider.py

```diff
@@ -200,16 +199,217 @@ def test_with_binary_media_types_in_swagger_and_on_resource(self):
+                                                           "/functions/${NoApiEventFunction.Arn}/invocations",
+                                            },
+                                            "responses": {},
+                                        },
+                                    }
+                                }
+
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        provider = ApiProvider(template)
+        route1 = Route(path='/path', method='GET', function_name='NoApiEventFunction')
+
+        self.assertIn(route1, provider.routes)
+        self.assertEquals(provider.api.stage_name, "dev")
+        self.assertEquals(provider.api.stage_variables, None)
+
+    def test_provider_stage_variables(self):
+        template = {
+            "Resources": {
+                "Stage": {
+                    "Type": "AWS::ApiGateway::Stage",
+                    "Properties": {
+                        "StageName": "dev",
+                        "Variables": {
+                            "vis": "data",
+                            "random": "test",
+                            "foo": "bar"
+                        },
+                        "RestApiId": "TestApi"
+                    }
+                },
+                "TestApi": {
+                    "Type": "AWS::ApiGateway::RestApi",
+                    "Properties": {
+                        "Body": {
+                            "paths": {
+                                "/path": {
+                                    "get": {
+                                        "x-amazon-apigateway-integration": {
+                                            "httpMethod": "POST",
+                                            "type": "aws_proxy",
+                                            "uri": {
+                                                "Fn::Sub": "arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31"
+                                                           "/functions/${NoApiEventFunction.Arn}/invocations",
+                                            },
+                                            "responses": {},
+                                        },
+                                    }
+                                }
+
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        provider = ApiProvider(template)
+        route1 = Route(path='/path', method='GET', function_name='NoApiEventFunction')
+        self.assertIn(route1, provider.routes)
+        self.assertEquals(provider.api.stage_name, "dev")
+        self.assertEquals(provider.api.stage_variables, {
+            "vis": "data",
+            "random": "test",
+            "foo": "bar"
+        })
+
+    def test_multi_stage_get_all(self):
+        template = OrderedDict({
```

Comment:
Can we just statically define this like the other tests? This is much harder to read and understand what the content of `template` is. Also remains consistent with all the other tests we have here.